src/Test/DatabaseResetter.php

```diff
@@ -53,27 +75,18 @@ public static function resetDatabase(KernelInterface $kernel): void
                 '--connection' => $connection,
             ]);
         }
-
-        self::createSchema($application, $registry);
-
-        self::$hasBeenReset = true;
-    }
-
-    public static function resetSchema(KernelInterface $kernel): void
-    {
-        $application = self::createApplication($kernel);
-        $registry = $kernel->getContainer()->get('doctrine');
-
-        self::dropSchema($application, $registry);
-        self::createSchema($application, $registry);
     }
 
     private static function createSchema(Application $application, ManagerRegistry $registry): void
     {
-        foreach (self::objectManagersToReset($registry) as $manager) {
-            self::runCommand($application, 'doctrine:schema:create', [
-                '--em' => $manager,
-            ]);
+        if (self::isResetUsingMigrations()) {
+            self::runCommand($application, 'doctrine:migrations:migrate', ['-n' => true]);
+        } else {
+            foreach (self::objectManagersToReset($registry) as $manager) {
+                self::runCommand($application, 'doctrine:schema:create', [
+                    '--em' => $manager,
+                ]);
+            }
         }
 
         if (!Factory::isBooted()) {
```

Comment:
I guess this code has to be moved to `resetDatabase()`? It strictly has nothing to do with what the method name is suggesting (`createSchema`)

Reply:
This is part of the `TestState::flushGlobalState();` below (which requires foundry to be booted). I think it should be kept together (but this is all changing in #153 so I'll re-evaluate there.)